test/test_find.cpp

```diff
@@ -23,7 +23,9 @@
     printf("Expected: %zu Actual: %zu\n", expected, actual); \
 } while (0)
 #else
-#define LOG(X, arg) {}
+#define LOG(X, arg) do { \
+    (void)arg; \
```

Comment:
@clalancette , I don't figure it out why `(void) arg` appear there. In my opinion, it makes sense like that:
```
#define LOG(X, arg) do { \
        ;
    } while (0)
```

Reply:
or save some space :-P
```
#define LOG(X, arg) do {;} while (0)
```

Reply:
TL;DR I think it is to avoid "unused variable warnings".

As a lot of this testing code is initializing variables without using them for anything else than the LOG call, if the LOG macro expands to nothing, the resulting code would have many unused variables. I can imagine cases where casting `X` as well would be relevant. This code just happens to always pass litterals as the first argument, but it should compile without warnings even if we pass a variable as the first argument. So I'd recommend to update this patch to void-cast X as well.

Though I didnt try to compile that branch with or without the cast so their may be another reason for this change.

Reply:
Yep, @mikaelarguedas is exactly right.  If you don't have the `(void)arg;`, then when you compile with `ENABLE_LOGGING` set to 0, you get a bunch of warnings about unused variables.  While this isn't fatal, I figured I may as well clean it up while I was in there.

Reply:
Thank for explanations.